web/gulpfile.babel.js

```diff
@@ -37,6 +38,13 @@ gulp.task('test', (done) => {
 
 gulp.task('server-hot', bg('node', './webpack/server'));
 
-gulp.task('server', ['set-dev-environment', 'server-hot'], bg('./node_modules/.bin/nodemon', './src/server'));
+gulp.task('nodemon', () => {
+  nodemon({
+    script: './src/server',
+    env: {'NODE_ENV': process.env.NODE_ENV || 'development'}
```

Comment:
Do you think `env` is needed here? `set-dev-environment` will set up NODE_ENV flag in development mode which should be enough (given @steida loves throwing when no NODE_ENV is present which is somehow a good practice)